extide/gradle/netbeans-gradle-tooling/src/main/java/org/netbeans/modules/gradle/tooling/NbProjectInfoBuilder.java

```diff
@@ -1562,6 +1563,23 @@ private void detectDependencies(NbProjectInfoModel model) {
         boolean ignoreUnresolvable = (project.getPlugins().hasPlugin("java-platform") &&
             Boolean.TRUE.equals(getProperty(project, "javaPlatform", "allowDependencies")));
 
+        // https://github.com/apache/netbeans/issues/8764
+        Function<ProjectDependency, Project> projDependencyToProject =
+            sinceGradleOrDefault(
+                "9.0",
+                () -> {
+                    Method getPath = ProjectDependency.class.getMethod("getPath");
+                    return dep -> {
+                        try {
+                            String path = (String) getPath.invoke(dep);
+                            return project.findProject(path);
+                        } catch (ReflectiveOperationException e) {
+                            throw new UnsupportedOperationException(e);
+                        }
+                    };
+                },
+                () -> ProjectDependency::getDependencyProject); // removed in Gradle 9
```

Comment:
IMO: this would be more future proof if we would call the new method directly and use reflection for the removed method. (basically the other way around)

@lkishalmi @sdedic do you think we can bump the gradle tooling version of this module so that we can call `getPath()` directly?

Reply:
Gradle >= 8.11 should have both APIs

Reply:
I have an alternative branch where I think I've done the necessary work to bump this module up to Gradle 8.11.1.  Tested locally on some personal Gradle projects using 7.5.x, 8.11-14, and 9.0.0 and they all open w/out error notifications. In the process there was a fun roadbump that turns out to be a JDK bug re: method references under certain conditions.

The 8.11.1 upgrade has several more required changes compared to this spot fix.  LMK how you'd like me to proceed.